corehq/apps/sms/management/commands/run_sms_queue.py

```diff
@@ -43,6 +54,8 @@ def create_tasks(self):
             if queued_sms.domain and skip_domain(queued_sms.domain):
                 continue
 
+            if settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS and not can_send_sms_now():
```

Comment:
why not do this outside the loop? what happens to these messages, will the get queued later?

Reply:
this is to stop the processing if we move ahead of the window for the day during the processing.
Consider a case where we fetched lots of queued SMS at 8:55 PM and kept processing it way after 9pm (where the window finishes), this change stops processing at 9pm.
The SMSes that were not picked, will get picked in next iteration.

Its not needed outside the loop here, its already covered when we call create_tasks periodically.